Improbable/Stdlib/Improbable.Stdlib/WorkerConnection.cs

```diff
@@ -125,9 +148,18 @@ private static Task<WorkerConnection> ConnectAsync(ILocatorOptions options, Conn
                     using (var locator = new Locator(options.SpatialOsHost, options.SpatialOsPort, locatorParameters))
                     using (var future = locator.ConnectAsync(connectionParameters))
```

Comment:
Some common code for polling the future that can be factored out. 


We might actually be able to write a generic extension method to transform it into a task.


```csharp
        public static Task<T> ToTask<T>(this Future<T> future, CancellationToken cancellation = default)
        {
            var tcs = new TaskCompletionSource<T>();
            cancellation.Register(() => tcs.TrySetCanceled(cancellation));

            Task.Run(() =>
            {
                while (true)
                {
                    cancellation.ThrowIfCancellationRequested();

                    if (future.TryGet(out var value, 50))
                    {
                        tcs.TrySetResult(value);
                        break;
                    }
                }
            }, cancellation);

            return tcs.Task;
        }
```

Or something like that

Reply:
Nice, good idea